cli/tools/compile.rs

```diff
@@ -53,16 +53,6 @@ pub async fn compile(
     );
   }
 
-  if cli_options.unstable_detect_cjs() {
```

Comment:
Is this on purpose? Shouldn't we land https://github.com/denoland/deno/pull/26439 first?

Reply:
Yes. It works in deno compile now so I got rid of the warning saying it doesn't work.